### PR TITLE
Show usage in total and by sid

### PR DESCRIPTION
**Pull request by iandyh**

### CODE REVIEW

File: shibuya/model/usage.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	collectionsToProjects := makeCollectionsToProjects(history)
+	uu := UnitUsage{
+		TotalVUH: make(map[string]float64),
+	}
+	sidHistory := []*CollectionLaunchHistory{}
+	s := &OwnerUsageSummary{
+		UnitUsage: uu,
+	}
+	for _, h := range history {
+		owner := h.Owner
```
Comment (rsw-a) [inline shibuya/model/usage.go:163]: same comment as I did at GetUsageSummary func.

File: shibuya/model/project.go [hunk 2]
```diff
@@ -161,3 +161,43 @@ func (p *Project) GetPlans() ([]*Plan, error) {
 	}
 	return r, nil
 }
+
+func GetProjectsBySid(sid string) ([]*Project, error) {
```
Comment (rsw-a) [inline shibuya/model/project.go:165]: no usages across code base. no need to add this implementation atm.

File: shibuya/model/project.go [hunk 3]
```diff
@@ -161,3 +161,43 @@ func (p *Project) GetPlans() ([]*Plan, error) {
...
+		r = append(r, p)
+	}
+	err = rows.Err()
+	if err != nil {
+		return r, err
+	}
+	return r, nil
+}
+
+func (p *Project) UpdateSid(sid string) error {
```
Comment (rsw-a) [inline shibuya/model/project.go:191]: no usages across code base. no need to add this implementation atm.

File: shibuya/model/usage.go [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+			contacts = append(contacts, p.Owner)
+			s.Contacts[sid] = contacts
+		}
+	}
+	for _, h := range history {
+		totalVUH := uu.TotalVUH
+		vhByOwner := s.VUHByOnwer
+		owner := h.Owner
+		sid := owner
+		_, err := strconv.ParseInt(owner, 10, 32)
```
[the remark this answers is not in the document]
Comment (rsw-a) [reply shibuya/model/usage.go:116]: ah, I noticed that you've changed a value of owner from Project.owner to Project.sid at contorller. that's what transition period means, right?

have you considered to clean up database directly without this tricky implementaion?
1. create batch script to update owner column at collection_launch_history2
2. release shibuya new logic

OR the history table has TTL so old records will be wiped by a batch app anyway?

**Review by rsw-a — COMMENTED** [review]
@iandyh 
I've put some comments, please have a look.

File: shibuya/model/usage.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (iandyh) [reply shibuya/model/usage.go:116]: About no.1, the exiting data will be kept as ML. The reason to keep is that we the billing report is generated monthly and we are going to expect all the new entries to have SID as the owner. So in a month or so, we will no longer billed by ML.

File: shibuya/model/usage.go [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+	history := []*CollectionLaunchHistory{}
+	for rs.Next() {
+		lh := new(CollectionLaunchHistory)
+		rs.Scan(&lh.CollectionID, &lh.Context, &lh.Owner, &lh.Vu, &lh.StartedTime, &lh.EndTime)
+		history = append(history, lh)
+	}
+	return history, nil
+}
+
+func calBillingHours(startedTime, endTime time.Time) float64 {
```
[the remark this answers is not in the document]
Comment (iandyh) [reply shibuya/model/usage.go:49]: It's ok. We billed usage by hour(hour is the minimum unit). So even user uses for 1 second, we bill it 1 hour. I'll provide the spec in the Slack.

File: shibuya/model/usage.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (rsw-a) [reply shibuya/model/usage.go:116]: I see your point. what I mean is that ppl might confuse from this implementation based on the table schema and the change on controller. that's my point. can you add a comment to explain the background of that?

File: shibuya/model/usage.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (rsw-a) [reply shibuya/model/usage.go:49]: ok, then, please remove the comments, it's wrong info.

File: shibuya/model/usage.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (iandyh) [reply shibuya/model/usage.go:116]: Sure.

File: shibuya/model/usage.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (iandyh) [reply shibuya/model/usage.go:49]: well, 0.1 is just an example. Showing the `ceiling` logic. I'll add a bit more comment.

**Comment by iandyh** [comment]
@rsw-a Updated the PR with polished comments.

**Review by rsw-a — APPROVED** [review]
looks good to me!

**Pull request merged**